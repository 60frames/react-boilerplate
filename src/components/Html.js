import React, { PropTypes } from 'react';

function Html({ css, js, html, head, initialState }) {
    return (
        <html lang="en">
            <head>
                <meta charSet="utf-8" />
                <meta httpEquiv="x-ua-compatible" content="ie=edge,chrome=1" />
                {head.title.toComponent()}
                {head.meta.toComponent()}
                <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no" />
                <meta name="apple-mobile-web-app-capable" content="yes" />
                <meta name="msapplication-tap-highlight" content="no" />
                <link rel="apple-touch-icon" href="apple-touch-icon.png" />
                {head.link.toComponent()}
                {css ? (
                    <link rel="stylesheet" href={css} />
                ) : null}
            </head>
            <body>
                <div id="root" dangerouslySetInnerHTML={{
                    __html: html
                }} />
                <script dangerouslySetInnerHTML={{
                    __html: `window.__INITIAL_STATE__ = ${JSON.stringify(initialState)}`
                }} />
                <script src={js}></script>
            </body>
        </html>
    );
}

Html.propTypes = {
    css: PropTypes.string,
    js: PropTypes.string.isRequired,
    html: PropTypes.string,
    head: PropTypes.object.isRequired,
    initialState: PropTypes.object.isRequired
};

export default Html;
