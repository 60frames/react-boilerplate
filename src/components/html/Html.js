import React, { PropTypes } from 'react';
import serialize from 'serialize-javascript';

function Html({ css, mainJs, chunkJs, html, head, initialState }) {
    return (
        <html lang="en">
            <head>
                <meta charSet="utf-8" />
                <meta httpEquiv="x-ua-compatible" content="ie=edge,chrome=1" />
                {head.title.toComponent()}
                {head.meta.toComponent()}
                <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no" />
                <meta name="apple-mobile-web-app-capable" content="yes" />
                <meta name="msapplication-tap-highlight" content="no" />

                <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" />
                <link rel="icon" type="image/png" href="/favicon-32x32.png" sizes="32x32" />
                <link rel="icon" type="image/png" href="/favicon-16x16.png" sizes="16x16" />
                <link rel="manifest" href="/manifest.json" />
                <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5" />
                <meta name="apple-mobile-web-app-title" content="60fram.es" />
                <meta name="application-name" content="60fram.es" />
                <meta name="theme-color" content="#ffffff" />

                {head.link.toComponent()}
                {css ? (
                    <link rel="stylesheet" href={css} />
                ) : null}
            </head>
            <body>
                <div id="root" dangerouslySetInnerHTML={{
                    __html: html
                }} />
                <script dangerouslySetInnerHTML={{
                    __html: `window.process = ${serialize({
                        env: {
                            BROWSER: 'true',
                            REDUX_LOGGER: process.env.REDUX_LOGGER,
                            API_ENDPOINT: process.env.API_ENDPOINT
                        }
                    })}`
                }} />
                <script dangerouslySetInnerHTML={{
                    __html: `window.__INITIAL_STATE__ = ${serialize(initialState)}`
                }} />
                <script src="/bootstrap.js"></script>
                {chunkJs.map(js => <script key={js} src={`/${js}`}></script>)}
                <script src={`/${mainJs}`}></script>
            </body>
        </html>
    );
}

Html.propTypes = {
    css: PropTypes.string,
    // js: PropTypes.string.isRequired,
    html: PropTypes.string,
    head: PropTypes.object.isRequired,
    initialState: PropTypes.object.isRequired
};

export default Html;
