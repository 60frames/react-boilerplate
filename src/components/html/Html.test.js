/* eslint-env jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import Html from 'components/html/Html';

describe('components/html/Html', () => {

    it('renders correctly', () => {
        const component = renderer.create(
            <Html
                css="https://60fram.es/bundle.css"
                js="https://60fram.es/bundle.js"
                html={`
                    <div id="root">
                        Hello World.
                    </div>
                `}
                head={{
                    title: {
                        toComponent: () => '<title>title</title>'
                    },
                    meta: {
                        toComponent: () => '<meta />'
                    },
                    link: {
                        toComponent: () => '<link />'
                    }
                    }}
                initialState={{
                    initial: 'state'
                }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
