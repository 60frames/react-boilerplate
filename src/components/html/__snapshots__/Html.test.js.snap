exports[`components/html/Html renders correctly 1`] = `
<html
  lang="en">
  <head>
    <meta
      charSet="utf-8" />
    <meta
      content="ie=edge,chrome=1"
      httpEquiv="x-ua-compatible" />
    <title>title</title>
    <meta />
    <meta
      content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no"
      name="viewport" />
    <meta
      content="yes"
      name="apple-mobile-web-app-capable" />
    <meta
      content="no"
      name="msapplication-tap-highlight" />
    <link
      href="/apple-touch-icon.png"
      rel="apple-touch-icon"
      sizes="180x180" />
    <link
      href="/favicon-32x32.png"
      rel="icon"
      sizes="32x32"
      type="image/png" />
    <link
      href="/favicon-16x16.png"
      rel="icon"
      sizes="16x16"
      type="image/png" />
    <link
      href="/manifest.json"
      rel="manifest" />
    <link
      color="#5bbad5"
      href="/safari-pinned-tab.svg"
      rel="mask-icon" />
    <meta
      content="60fram.es"
      name="apple-mobile-web-app-title" />
    <meta
      content="60fram.es"
      name="application-name" />
    <meta
      content="#ffffff"
      name="theme-color" />
    <link />
    <link
      href="https://60fram.es/bundle.css"
      rel="stylesheet" />
  </head>
  <body>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "
                            <div id=\"root\">
                                Hello World.
                            </div>
                        ",
        }
      }
      id="root" />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.process = {
                                env: {
                                    BROWSER: \'true\',
                                    REDUX_LOGGER: \'undefined\',
                                    API_ENDPOINT: \'undefined\'
                                }
                            }",
        }
      } />
    <script
      dangerouslySetInnerHTML={
        Object {
          "__html": "window.__INITIAL_STATE__ = {\"initial\":\"state\"}",
        }
      } />
    <script
      src="https://60fram.es/bundle.js" />
  </body>
</html>
`;
